ashwing/amazon-kinesis-client
# [Cherry-pick to ltr_1] Add DataFetcher and ShardDetector interface changes (#9)

* Add DataFetcher and ShardDetector interface changes

* Allow passing in custom shard detector and custom data fetcher through configs

* update javadoc comment

* Add extra type check for KinesisShardDetector

* Update javadoc

* update prefetchRecordsPublisher

* ShardDetector now passed created in DynamoDBLeaseManagementFactory
DataFetcher interface updated
Updated Functional interface for ShardDetector and DataFetcher

* update var names

* Added DataFetcherProviderConfig to make functional interface arugments clear

* Deprecate getDataFetcher method

* Add additional vars for DataFetcherProviderConfig

* Add missing doc strings & use interface for Config impl

* Fix test

* tests & readability

* Remove constructor args from DDBLeaseManagementFactory
Code clean up
Update RetrievalConfig

* deprecate method, remove extra constructor

* Move dataFetcher to PollingConfig, add validation

* remove override method

* remove unused annotation + update exception type

* make lombok annotations explicit

* remove commented code

* Clean up SynchronousBlockingRetrievalFactory

* use custom shard detector

* update doc strings

Co-authored-by: Renju Radhakrishnan <rrenju@amazon.com>

*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
